2024/half-2/day-12/array-reduce/array-reduce.test.ts

```
@@ -0,0 +1,76 @@
+import { describe, expect, test } from "vitest";
+import "./array-reduce";
+
+const add = (prev: any, curr: any) => prev + curr;
+const multiplyByIndex = (prev: number, curr: number, index: number) =>
+  prev + curr * index;
+const subtract = (prev: number, curr: number) => prev - curr;
+const sumOfSquares = (prev: any, curr: any, index: number, array: Array<any>) =>
+  prev + curr * array[index];
+const combineObj = (prev: Object, curr: Object) => ({ ...prev, ...curr });
```

Biome

[error] Don't use 'Object' as a type.

Prefer explicitly define the object shape. This type means "any non-nullable value", which is slightly better than 'unknown', but it's still a broad type.

(lint/complexity/noBannedTypes)